addons/product/models/product_template.py

```diff
@@ -774,7 +791,38 @@ def _get_mapped_attribute_names(self, parent_combination=None):
             for attribute_value in all_product_attribute_values
         }
 
-    def _is_combination_possible(self, combination, parent_combination=None):
+    def _is_combination_possible_by_config(self, combination, ignore_no_variant=False):
+        """Return whether the given combination is possible according to the config of attributes on the template
+
+        :param combination: the combination to check for possibility
+        :type combination: recordset `product.template.attribute.value`
+
+        :return: wether the given combination is possible according to the config of attributes on the template
+        :rtype: bool
+        """
+        self.ensure_one()
+
+        valid_pta_line_ids = self.valid_product_template_attribute_line_ids
```

Comment:
Don't suffix the variable with `_ids` when it contains record sets and not ids.

Also usually in this code `pta_line` => `ptal`
That said I don't like the abbreviated name when it is plural.

Since this is a local variable maybe just name it `attribute_lines` just like in `_get_possible_combinations`.